### PR TITLE
Update DaprClient API

### PR DESCRIPTION
**Pull request by shubham1172**
# Description

Refactor checklist
- [x] Code changes
- [x] Tests
- [x] Examples
- [x] Documentation

## Issue reference

Please reference the issue this PR will close: #442

## Checklist

Please make sure you've completed the relevant tasks for this PR, out of the following list:

- [X] Code compiles correctly
- [ ] Created/updated tests
- [ ] Extended the documentation

### CODE REVIEW

File: daprdocs/content/en/js-sdk-docs/js-actors/_index.md [hunk 1]
```diff
@@ -136,7 +139,7 @@ import ParkingSensorInterface from "./ParkingSensorInterface";
 const daprHost = "127.0.0.1";
 const daprPort = "50000";
 
-const client = new DaprClient(daprHost, daprPort);
+const client = new DaprClient({ daprHost: daprHost, daprPort: daprPort });
```
Comment (XavierGeerinck) [inline daprdocs/content/en/js-sdk-docs/js-actors/_index.md:142]: Shouldn't we just write `({ daprHost, daprPort })` ?

File: daprdocs/content/en/js-sdk-docs/js-client/_index.md [hunk 2]
```diff
@@ -35,10 +35,14 @@ const serverHost = "127.0.0.1"; // App Host of this Example Server
 const serverPort = "50051"; // App Port of this Example Server
 
 // HTTP Example
-const client = new DaprClient(daprHost, daprPort);
+const client = new DaprClient({ daprHost: daprHost, daprPort: daprPort });
```
Comment (XavierGeerinck) [inline daprdocs/content/en/js-sdk-docs/js-client/_index.md:38]: Shouldn't we just write ({ daprHost, daprPort }) ?

File: daprdocs/content/en/js-sdk-docs/js-client/_index.md [hunk 3]
```diff
@@ -66,7 +70,11 @@ Since HTTP is the default, you will have to adapt the communication protocol to
 
 ```typescript
 import { DaprClient, CommunicationProtocol } from "@dapr/dapr";
-const client = new DaprClient(daprHost, daprPort, CommunicationProtocol.GRPC);
+const client = new DaprClient({
+  daprHost: daprHost,
```
Comment (XavierGeerinck) [inline daprdocs/content/en/js-sdk-docs/js-client/_index.md:74]: Shouldn't we just write ({ daprHost, daprPort }) ?

File: daprdocs/content/en/js-sdk-docs/js-client/_index.md [hunk 4]
```diff
@@ -134,7 +151,7 @@ const daprHost = "127.0.0.1";
 const daprPort = "3500";
 
 async function start() {
-  const client = new DaprClient(daprHost, daprPort);
+  const client = new DaprClient({ daprHost: daprHost, daprPort: daprPort });
```
Comment (XavierGeerinck) [inline daprdocs/content/en/js-sdk-docs/js-client/_index.md:154]: Shouldn't we just write ({ daprHost, daprPort }) ?

File: daprdocs/content/en/js-sdk-docs/js-client/_index.md [hunk 5]
```diff
@@ -231,7 +248,7 @@ start().catch((e) => {
 import { DaprClient } from "@dapr/dapr";
 
 async function start() {
-  const client = new DaprClient(daprHost, daprPort);
+  const client = new DaprClient({ daprHost: daprHost, daprPort: daprPort });
```
Comment (XavierGeerinck) [inline daprdocs/content/en/js-sdk-docs/js-client/_index.md:251]: Shouldn't we just write ({ daprHost, daprPort }) ?

File: daprdocs/content/en/js-sdk-docs/js-client/_index.md [hunk 6]
```diff
@@ -282,7 +299,7 @@ const daprHost = "127.0.0.1";
 const daprPort = "3500";
 
 async function start() {
-  const client = new DaprClient(daprHost, daprPort);
+  const client = new DaprClient({ daprHost: daprHost, daprPort: daprPort });
```
Comment (XavierGeerinck) [inline daprdocs/content/en/js-sdk-docs/js-client/_index.md:302]: Shouldn't we just write ({ daprHost, daprPort }) ?

File: daprdocs/content/en/js-sdk-docs/js-client/_index.md [hunk 7]
```diff
@@ -355,7 +372,7 @@ const daprHost = "127.0.0.1";
...
```
Comment (XavierGeerinck) [inline daprdocs/content/en/js-sdk-docs/js-client/_index.md:375]: Shouldn't we just write ({ daprHost, daprPort }) ?

File: daprdocs/content/en/js-sdk-docs/js-client/_index.md [hunk 8]
```diff
@@ -383,7 +400,7 @@ const daprHost = "127.0.0.1";
...
```
Comment (XavierGeerinck) [inline daprdocs/content/en/js-sdk-docs/js-client/_index.md:403]: Shouldn't we just write ({ daprHost, daprPort }) ?

File: daprdocs/content/en/js-sdk-docs/js-client/_index.md [hunk 9]
```diff
@@ -438,7 +455,7 @@ const daprHost = "127.0.0.1";
 const daprPortDefault = "3500";
 
 async function start() {
-  const client = new DaprClient(daprHost, daprPort);
+  const client = new DaprClient({ daprHost: daprHost, daprPort: daprPort });
```
Comment (XavierGeerinck) [inline daprdocs/content/en/js-sdk-docs/js-client/_index.md:458]: Shouldn't we just write ({ daprHost, daprPort }) ?

File: daprdocs/content/en/js-sdk-docs/js-logger/_index.md [hunk 10]
```diff
@@ -20,7 +20,10 @@ There are five levels of logging in **descending order of importance** - `error`
 import { CommunicationProtocolEnum, DaprClient, LogLevel } from "@dapr/dapr";
 
 // create a client instance with log level set to verbose.
-const client = new DaprClient(daprHost, daprPort, CommunicationProtocolEnum.HTTP, {
+const client = new DaprClient({
+  daprHost: daprHost,
```
Comment (XavierGeerinck) [inline daprdocs/content/en/js-sdk-docs/js-logger/_index.md:24]: Shouldn't we just write ({ daprHost, daprPort }) ?

File: examples/grpc/hello-world-distributed/client/src/index.ts [hunk 11]
```diff
@@ -18,7 +18,7 @@ const daprPort = "3500";
 const daprAppId = "my-server";
 
 async function start() {
-  const client = new DaprClient(daprHost, daprPort);
+  const client = new DaprClient({ daprHost: daprHost, daprPort: daprPort });
```
Comment (XavierGeerinck) [inline examples/grpc/hello-world-distributed/client/src/index.ts:21]: Shouldn't we just write ({ daprHost, daprPort }) ?

File: src/actors/runtime/AbstractActor.ts [hunk 12]
```diff
@@ -54,12 +54,12 @@ export default abstract class AbstractActor {
   constructor(daprClient: DaprClient, id: ActorId) {
     this.daprClient = daprClient;
     this.actorClient = new ActorClient(
-      daprClient.getDaprHost(),
-      daprClient.getDaprPort(),
-      daprClient.getCommunicationProtocol(),
-      daprClient.getOptions(),
+      daprClient.options.daprHost,
```
Comment (XavierGeerinck) [inline src/actors/runtime/AbstractActor.ts:57]: Is options always set? Wouldn't `daprClient?.options?.` be better?

File: src/implementation/Client/GRPCClient/actor.ts [hunk 13]
```diff
@@ -30,10 +30,10 @@ export default class GRPCClientActor implements IClientActorBuilder {
   create<T>(actorTypeClass: Class<T>): T {
     const builder = new ActorProxyBuilder<T>(
       actorTypeClass,
-      this.client.getClientHost(),
-      this.client.getClientPort(),
-      this.client.getClientCommunicationProtocol(),
-      this.client.getOptions(),
+      this.client.options.daprHost,
```
Comment (XavierGeerinck) [inline src/implementation/Client/GRPCClient/actor.ts:33]: is `client.options.` always set?

File: test/e2e/common/client.test.ts [hunk 14]
```diff
@@ -26,15 +26,21 @@ describe("common/client", () => {
   let grpcClient: DaprClient;
 
   beforeAll(async () => {
-    httpClient = new DaprClient(daprHost, daprHttpPort, CommunicationProtocolEnum.HTTP, {
+    httpClient = new DaprClient({
+      daprHost: daprHost,
```
Comment (XavierGeerinck) [inline test/e2e/common/client.test.ts:30]: Shouldn't we just write ({ daprHost, daprPort }) ?

File: test/e2e/common/client.test.ts [hunk 15]
```diff
@@ -26,15 +26,21 @@ describe("common/client", () => {
...
+      daprPort: daprHttpPort,
+      communicationProtocol: CommunicationProtocolEnum.HTTP,
       logger: loggerSettings,
     });
-    await httpClient.getDaprClient().getClient();
+    await httpClient.daprClient.getClient();
 
-    grpcClient = new DaprClient(daprHost, daprGrpcPort, CommunicationProtocolEnum.GRPC, {
+    grpcClient = new DaprClient({
+      daprHost: daprHost,
```
Comment (XavierGeerinck) [inline test/e2e/common/client.test.ts:38]: Shouldn't we just write ({ daprHost, daprPort }) ?

File: test/e2e/grpc/client.test.ts [hunk 16]
```diff
@@ -31,7 +31,10 @@ describe("grpc/client", () => {
   // We need to start listening on some endpoints already
   // this because Dapr is not dynamic and registers endpoints on boot
   beforeAll(async () => {
-    client = new DaprClient(daprHost, daprPort, CommunicationProtocolEnum.GRPC, {
+    client = new DaprClient({
+      daprHost: daprHost,
```
Comment (XavierGeerinck) [inline test/e2e/grpc/client.test.ts:35]: Shouldn't we just write ({ daprHost, daprPort }) ?

**Review by XavierGeerinck — CHANGES_REQUESTED** [review]
Awesome PR! Tbh one small remark: let's replace `({ daprHost: daprHost, daprPort: daprPort })` with `({ daprHost, daprPort })` ?

File: src/implementation/Client/GRPCClient/actor.ts (hunk 13 above)
Comment (shubham1172) [reply src/implementation/Client/GRPCClient/actor.ts:33]: Yup, if you look at GRPCClient or HTTPClient ctor, it always takes an option and sets it. The DaprClient class takes a partial option, and creates the entire object using defaults.

File: src/actors/runtime/AbstractActor.ts (hunk 12 above)
Comment (shubham1172) [reply src/actors/runtime/AbstractActor.ts:57]: See below.

**Review by XavierGeerinck — APPROVED** [review]

**Pull request merged**